proposals/lip-introduce_sidechain_recovery_mechanism.md

```diff
@@ -0,0 +1,555 @@
+*   Let `process` be the internal function with the same name defined in [Introduce cross-chain messages LIP][CCM].
+*   Let `unescrow` be the function with the same name defined in [Define state and state transitions of Token module LIP][tokenCCM].
+*   Let `RMTVerify` be the `verifyDataBlock` function specified in [the appendix C of LIP 0031][LIP31appendixC].
+*   Let `RMTCalculateRoot` be the `calculateRootFromUpdateData` function specified in [the appendix E of LIP 0031][LIP31appendixE].
+*   Let `SMTVerify` be the `verify` function specified in the [LIP 39][SMTLIP].
+*   Let `SMTCalculateRoot` be the `calculateRoot` function specified in the [LIP 39][SMTLIP].
+
+
+### Message Recovery Command 
+
+The command ID is `COMMAND_ID_MESSAGE_RECOVERY`.
+
+* `params` property:
+    * `chainID`: An integer representing the chain ID of the terminated sidechain.
+    * `crossChainMessages`: An array of serialized CCMs, according to the schema specified in [Cross-chain messages LIP][CCMschema], to be recovered.
+    * `idxs`: An array of indices corresponding to the position in the Merkle tree of the outbox root of the sidechain for the elements in `crossChainMessages` as specified in [LIP 0031][LIP31inclusions].
+    * `siblingHashes`: Array of bytes with the paths in the Merkle tree for the proofs of inclusion of `crossChainMessages` in the outbox root of the sidechain as specified in [LIP 0031][LIP31inclusions].
+
+#### Message Recovery Command Schema
+
+```java
+messageRecoveryParams = {
+    "type": "object",
+    "required": ["chainID", "crossChainMessages", "idxs", "siblingHashes"],
+    "properties": {
+        "chainID": {
+            "dataType": "uint32",
+            "fieldNumber": 1
+        },
+        "crossChainMessages": {
+            "type": "array",
+            "items": {
+                "dataType": "bytes"
+            },
+            "fieldNumber": 2
+        },
+        "idxs": {
+            "type": "array",
+            "items": {
+                "dataType": "uint64"
+            },
+            "fieldNumber": 3
+        },
+        "siblingHashes": {
+            "type": "array",
+            "items": {
+                "dataType": "bytes"
+            },
+            "fieldNumber": 4
+        }
+    }
+}
+```
+
+#### Message Recovery Command Verification
+
+Let `trs` be a transaction with module ID `MODULE_ID_INTEROPERABILITY` and command ID `COMMAND_ID_MESSAGE_RECOVERY` to be verified.
+Also, let `deserializedCCMs` be an array with the deserialization of every element in `trs.params.crossChainMessages` according to the schema specified in [Cross-chain messages LIP][CCMschema].
+Then the set of validity rules to validate `trs.params` are:
+
+```python
+if trs.params.chainID does not correspond to a registered sidechain:
+    return false
+sidechainAccount = account(trs.params.chainID)
+
+# chain has to be either terminated or inactive
+if sidechainAccount.status != CHAIN_TERMINATED and isLive(trs.params.chainID, timestamp):
```

Comment:
`timestamp` is not defined so far. Maybe add something like `timestamp = timestamp of the block including trs` above (or something like that)

Reply:
added as suggested